cms/db/filecacher.py

```diff
@@ -543,6 +545,33 @@ def _create_directory_or_die(directory):
             logger.error(msg)
             raise RuntimeError(msg)
 
+    def precache_lock(self):
+        """Lock the (shared) cache for precaching if it is currently unlocked.
+
+        The purpose is to avoid wasting resources by ensuring that on each
+        machine, only one worker precaches at any time.
+
```

Comment:
Would you like to also document here that normal cache operations can be called while the lock is held (including even from the same `FileCacher` in a different greenlet)?

Reply:
Done. I clarified that locking is optional and just an optimization.